Lib/test/test_httplib.py

```diff
@@ -1641,6 +1641,74 @@ class ExtendedReadTestContentLengthKnown(ExtendedReadTest):
     _header, _body = ExtendedReadTest.lines.split('\r\n\r\n', 1)
     lines = _header + f'\r\nContent-Length: {len(_body)}\r\n\r\n' + _body
 
+    def _test_incomplete_read(self, read_meth, expected_none):
+        resp = self.resp
+        # Reduce the size of content the response object will read to
+        # cause the incomplete read.
+        resp.fp.read(1)
+        with self.assertRaises(client.IncompleteRead) as cm:
+            while True:
+                data = read_meth()
+                if not data:
+                    break
+        exception = cm.exception
+        # Unlike `read1` and `readline`, `read` tries to read the whole
+        # content during one call, so it's partial is not empty in this
+        # case.
+        # `read1` and `readline` raise `IncompleteRead` only when they
+        # read 0 bytes before all expected content has been read, so the
+        # partial is empty.
+        if read_meth == self.resp.read:
+            expected_partial = self._body[1:].encode()
+        else:
+            expected_partial = b""
+        self.assertEqual(exception.partial, expected_partial)
+        if expected_none:
+            self.assertIsNone(exception.expected)
+        else:
+            self.assertEqual(exception.expected, 1)
```

Comment:
Why not call with say expected=None or expected=1 and then just
```suggestion
        if self.assertEqual(exception.expected, expected)
```

Reply:
I did this this way because `_test_incomplete_read` itself specified how many bytes to miss, so passing `expected=1` to it would make actual test methods more coupled.
But `_test_incomplete_read` wasn't aware whether for `expected` of a specific read method (`read`/`read1`/`readline`) was always set to `None`